sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -18,26 +31,169 @@ export class AssessmentController {
   ) {}
 
   /**
-   * This is a sample code to demonstrated the workers feature.
-   * @param job information to execute the work.
-   * @returns job completion or job failure information.
+   * Loads consolidated assessment including the assessment itself, application
+   * dynamic data, supporting users, income and more. Includes the ability to filter
+   * only the necessary data using jsonpath expressions (@see https://goessner.net/articles/JsonPath).
+   * The filtered data must ensures that no PII (Personal Identifiable Information)
+   * data will be sent to the workflow.
+   * The filter is executed based on a key/value pair containing the jsonpath expression
+   * received through the job custom headers.
+   * @returns filtered consolidated information.
    */
-  @ZeebeWorker("load-assessment-data")
+  @ZeebeWorker("load-assessment-data", { fetchVariable: [ASSESSMENT_ID] })
   async loadAssessmentData(
     job: Readonly<
-      ZeebeJob<
-        AssessmentDataWorkerInDTO,
-        ICustomHeaders,
-        AssessmentDataWorkerOutDTO
-      >
+      ZeebeJob<AssessmentDataWorkerInDTO, ICustomHeaders, IOutputVariables>
     >,
   ): Promise<MustReturnJobActionAcknowledgement> {
     const assessment = await this.studentAssessmentService.getById(
       job.variables.assessmentId,
     );
     if (!assessment) {
-      return job.fail("Assessment not found.");
+      return job.error(ASSESSMENT_NOT_FOUND, "Assessment not found.");
+    }
+    const assessmentDTO = this.createAssessmentDTO(assessment);
+    const outputVariables = filterObjectProperties(
+      assessmentDTO,
+      job.customHeaders,
+    );
+    return job.complete(outputVariables);
+  }
+
+  /**
+   * Creates a well-known object that represents the universe of possible
+   * information that can be later filtered.
+   * @param assessment assessment to be converted.
+   * @returns well-known object that represents the universe of possible
+   * information that can be later filtered.
+   */
+  private createAssessmentDTO(
```

Comment:
When looking at this method name initially it looked like it creates assessment. Just a suggestion to rename as transformTOAssessmentDTO or buildAssessmentDTO.